### PR TITLE
.Net: Polishing KernelSyntaxExample comments

### PR DESCRIPTION
**Pull request by teresaqhoang**
### Motivation and Context

This PR adds some more descriptive comments on set-up and expected results for 2 examples.

Also fixes a bug in the README by changing `ACS` -> `AzureAISearch`

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by glahaye — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/README.md [hunk 1]
```diff
@@ -238,10 +239,10 @@ Once the token is acquired, it is added to the HTTP authentication header via th
 
 To construct this provider, the caller must specify:
 
-- _Client ID_ � identifier of the calling application. This is acquired by [registering your application with the Microsoft Identity platform](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
-- _Tenant ID_ � identifier of the target service tenant, or �common�
-- _Scopes_ � permissions being requested
-- _Redirect URI_ � for redirecting the user back to the application. (When running locally, this is typically http://localhost.)
+- _Client ID_ – identifier of the calling application. This is acquired by [registering your application with the Microsoft Identity platform](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
```
Comment (TaoChenOSU) [inline dotnet/samples/KernelSyntaxExamples/README.md:242]: What is this character?

File: dotnet/samples/KernelSyntaxExamples/Example54_AzureChatCompletionWithData.cs [hunk 2]
```diff
@@ -8,19 +8,25 @@
 
 /**
  * This example shows how to use Azure OpenAI Chat Completion with data.
- * More information: <see href="https://learn.microsoft.com/en-us/azure/ai-services/openai/use-your-data-quickstart"/>
  */
 public static class Example54_AzureChatCompletionWithData
 {
+    /* Set-up
```
Comment (RogerBarreto) [inline dotnet/samples/KernelSyntaxExamples/Example54_AzureChatCompletionWithData.cs:14]: Suggest updating all the comments to use the `/// <summary>` approach.

File: dotnet/samples/KernelSyntaxExamples/README.md (hunk 1 above)
Comment (teresaqhoang) [reply dotnet/samples/KernelSyntaxExamples/README.md:242]: No idea, I saw this too and tried to revert it but it didn't work :/ Let me try to commit the suggestion from here

Comment (teresaqhoang) [reply dotnet/samples/KernelSyntaxExamples/README.md]: I think it was the emdash but should be fixed now!

Comment (teresaqhoang) [reply dotnet/samples/KernelSyntaxExamples/README.md]: Looks like it has something to do with the encoding? Half the time it renders in GH, the other half it doesn't
image
image

File: dotnet/samples/KernelSyntaxExamples/Example54_AzureChatCompletionWithData.cs (hunk 2 above)
Comment (teresaqhoang) [reply dotnet/samples/KernelSyntaxExamples/Example54_AzureChatCompletionWithData.cs]: Updated to XML function comments

**Review by crickman — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**